assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditBuildConfigController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditBuildConfigController', function ($scope, $routeParams, DataService, ProjectsService, $filter, ApplicationGenerator, Navigate, $location, AlertMessageService, SOURCE_URL_PATTERN) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.buildConfig = null;
+    $scope.alerts = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.sourceURLPattern = SOURCE_URL_PATTERN;
+    $scope.options = {};
+    $scope.buildFromTypes = [
+      {
+        "id": "ImageStreamTag",
+        "title": "Image Stream Tag"
+      },
+      {
+        "id": "DockerImage",
+        "title": "Docker Image Link"
+      }
+    ];
+    $scope.pushToTypes = [
+      {
+        "id": "ImageStreamTag",
+        "title": "Image Stream Tag"
+      },
+      {
+        "id": "DockerImage",
+        "title": "Docker Image Link"
+      },
+      {
+        "id": "None",
+        "title": "--- None ---"
+      }
+    ];
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },
+      {
+        title: "Edit"
+      }
+    ];
+    $scope.buildFrom = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    };
+    $scope.pushTo = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    };
+    $scope.sources = {
+      "binary": false,
+      "dockerfile": false,
+      "git": false,
+      "image": false,
+      "contextDir": false
+    };
+    $scope.triggers = {
+      "webhook": false,
+      "imageChange": false,
+      "configChange": false
+    };
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+    var watches = [];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        DataService.get("buildconfigs", $routeParams.buildconfig, context).then(
+          // success
+          function(buildConfig) {
+            $scope.buildConfig = buildConfig;
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
+            $scope.strategyType = $scope.buildConfig.spec.strategy.type;
+            $scope.envVars = $filter('envVarsPair')($scope.buildStrategy.env);
+
+            angular.forEach($scope.buildConfig.spec.triggers, function(value, key) {
```

Comment:
Wonder if this logic could be pushed up into a service.  Perhaps something like:

```javascript
$scope.triggers = BuildConfigService.getTriggers(buildConfig);
```